clients/src/main/java/org/apache/kafka/common/internals/PartitionStates.java

```diff
@@ -52,10 +59,12 @@ public void moveToEnd(TopicPartition topicPartition) {
     public void updateAndMoveToEnd(TopicPartition topicPartition, S state) {
         map.remove(topicPartition);
         map.put(topicPartition, state);
+        size = map.size();
```

Comment:
Nit: do we want to have a `updateSize` method that does this and is used in every method?